pontoon/sync/formats/l20n.py

```diff
@@ -0,0 +1,183 @@
+            self.structure = L20nParser().parse(resource.read())
+            order = 0
+            comments = []
+            for obj in self.structure.body:
+                if obj.type == 'Entity':
+                    if obj.value.type == 'String':
+                        self.entities[obj.id.name] = L20NEntity(
+                            obj.id.name,
+                            obj.value.source,
+                            '',
+                            {None: obj.value.source},
+                            comments,
+                            order
+                        )
+                        order += 1
+
+                    elif obj.value.type == 'Hash':
+                        strings = {
+                            self.locale.get_plural_index(item.id.name):
+                            item.value.source for item in obj.value.items
+                        }
+                        self.entities[obj.id.name] = L20NEntity(
+                            obj.id.name,
+                            strings.get(0, ''),
+                            strings.get(1, ''),
+                            strings,
+                            comments,
+                            order
+                        )
+                        order += 1
+
+                    for attr in obj.attrs:
+                        key = '.'.join([obj.id.name, attr.id.name])
+                        self.entities[key] = L20NEntity(
+                            key,
+                            attr.value.source,
+                            '',
+                            {None:attr.value.source},
+                            order=order
+                        )
+                        order += 1
+                    comments = []
+
+                elif obj.type == 'Comment':
+                    comments.append(obj.body)
+
+                else:
+                    comments = []
+
+    @property
+    def translations(self):
+        return sorted(self.entities.values(), key=lambda e: e.order)
+
+    def save(self, locale):
+        if not self.source_resource:
+            raise SyncError('Cannot save l20n resource {0}: No source resource given.'
+                            .format(self.path))
+
+        with codecs.open(self.source_resource.path, 'r', 'utf-8') as resource:
+            new_structure = L20nParser().parse(resource.read())
+
+        to_remove = []
+        for res_entity in new_structure.body:
+            if res_entity.type == 'Entity':
+                entity_id = res_entity.id.name
+                translations = self.entities[entity_id].strings
+                comments = self.entities[entity_id].comments
+
+                if not translations:
+                    entity_idx = new_structure.body.index(res_entity)
+
+                    for comment_entity in new_structure.body[entity_idx-len(comments):entity_idx]:
+                        to_remove.append(comment_entity)
```

Comment:
Change this to something like: `to_remove.append((new_structure.body, comment_entity))` or you get `TypeError: 'Comment' object is not iterable`.